content/articles/nodejs-rate-limiting/index.md

```diff
@@ -0,0 +1,248 @@
+### Node.js Rate Limiting
+
+The rate-limiting feature makes it possible to secure the Backend API from malicious attacks. It allows us to cap the number of requests that a user can make to our APIs. Rate limiting is widely used by API providers to limit the number of requests an unsubscribed user can make within a given duration i.e. [https://newsapi.org](https://newsapi.org/pricing) limits users with developer accounts only to make 100 requests per day.
+
+### Prerequisites
+
+1. A General understanding of HTTP requests/responses.
+2. [Node.js](https://nodejs.org/en/) installed on your computer.
+3. Some knowledge of [Node.js](https://www.w3schools.com/nodejs/).
+
+Rate limiting is a feature used to control outgoing and incoming requests in a server. We might limit the number of requests a user without a premium account makes to 100 to an hour. When a user makes more requests than the provided limit in the window duration then an error message is returned informing the user that the allowed limit is exceeded.
+
+#### Project setup
+
+1. Create a folder named `limiter` for the project.
+2. In the `limiter` directory created above, execute the command below to initialize a Node.js project.
+   ```bash
+   $ npm init
+   ```
+3. Install `express` in the project directory.
+   ```bash
+   $ npm install --save express
+   ```
+4. In the project directory, create the entry point file named `index.js` add the code snippet below.
+
+   ```javascript
+   const express = require("express");
+   const app = express();
+   const port = 3000;
+   //returns the string Hello World when / is visited
+   app.get("/", (req, res) => {
+     res.send("Hello World!");
+   });
+
+   app.listen(port, () => {
+     console.log(`Example app listening at http://localhost:${port}`);
+   });
+   ```
+
+5. Execute the command below in the project directory to ensure that our application runs with no errors.
+
+   ```bash
+   $ node index.js
+   ```
+
+The provided code contains an API that returns a list of books when we send a `GET` request to the `/posts` endpoint. We are going to implement a rate limiter that restricts API access to a given number of requests within a specied duration. We are going to use a middleware layer to implement the rate limiter.
```

Comment:
I think there is a missing section in this place. You explain that you have implemented rate-limiting with IP address but there is no code snippet available. Please check on this. :eyes: